examples/vision/retinanet.py

```diff
@@ -0,0 +1,700 @@
+"""
+Title: Object Detection with RetinaNet
+Author: [Srihari Humbarwadi](https://twitter.com/srihari_rh)
+Date created: 2020/05/17
+Last modified: 2020/06/22
+Description: Implementation of Focal Loss for Dense Object Detection.
+"""
+import os
+import re
+import zipfile
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+
+import matplotlib.pyplot as plt
+import tensorflow_datasets as tfds
+
+"""
+## Download subset of COCO2017 dataset
+"""
+
+url = "https://github.com/srihari-humbarwadi/datasets/releases/download/v0.1.0/data.zip"
+filename = os.path.join(os.getcwd(), "data.zip")
+tf.keras.utils.get_file(filename, url, extract=True, archive_format="zip")
+
+
+with zipfile.ZipFile("data.zip", "r") as z_fp:
+    z_fp.extractall("./")
+
+"""
+## Utility functions
```

Comment:
There are many classes, utility functions, etc. being defined in this example. Make sure to add a few sentences to introduce the role of each one. Otherwise the example will read like a long list of objects parachuted with no context.